public/javascripts/notebook/observable.js

```diff
@@ -113,31 +113,47 @@ return new function () {
 
 
 
-  this.makeObservableHelper = function (id, kind, initialValue) {
+  this.makeObservableHelper = function (id, kind, initialValue, df) {
+    var _df = df
+    if (_.isUndefined(_df)) {
+      _df = function(publishValue) {
+        this.df.done(function(ob) {
+          ob(publishValue);
+        });
+      }
+      _df.df = $.Deferred();
+      _df.subscribe = function(subscriber) {
+        this.df.done(function(ob) {
+          ob.subscribe(subscriber);
+        });
+      };
+    }
     if(!this.isInitialized()) {
       // in case the Observable isn't initialized, we differ the creation of the observable
       var args = Array.prototype.slice.call(arguments);
       var me = this;
+      var current_df = _df;
       console.warn("Delaying the creation of Observable (" + args.join(",") + ") since Observable isn't initialized yet");
       console.log("Observable not initialized:", me);
       events.on('Observable.ready', function() {
-        me.makeObservableHelper.apply(me, args);
+        me.makeObservableHelper.apply(me, args, current_df);
```

Comment:
hmm does `.apply()` take 3 args? so it should be `apply(this, args)`.
thus we just need to replace the last argument this the deferred `current_df`.